assets/src/components/v2/departures/later_departures.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import _ from "lodash";
+import cx from "classnames";
+import weakKey from "weak-key";
+
+import { hasOverflowX } from "Util/util";
+
+import DepartureRow from "./departure_row";
+import { Row } from "./normal_section";
+import RoutePill from "./route_pill";
+
+const MAX_LATER_DEPARTURES = 5;
+
+const LaterDepatures = ({ rows }: { rows: Row[] }) => {
+  const [limit, setLimit] = useState(MAX_LATER_DEPARTURES);
+  const departures = useMemo(
+    () =>
+      _.take(
+        rows.filter(
+          (r): r is DepartureRow & { type: "departure_row" } =>
+            r.type === "departure_row",
+        ),
+        limit,
+      ),
+    [rows, limit],
+  );
+
+  const ref = useRef<HTMLDivElement>(null);
+  const [idx, setIdx] = useState(0);
+
+  useLayoutEffect(() => {
+    if (hasOverflowX(ref)) {
+      setLimit(Math.max(0, limit - 1));
+    }
+  }, [limit]);
+
+  useEffect(() => {
+    if (departures.length < 2) return;
+    const refreshMs = Math.min(3750, 15000 / departures.length);
```

Comment:
Can we extract the `15000` into a descriptively-named constant and calculate the `3750` from it? It'd be nice if we could actually use whatever the configured refresh rate of the screen is, but maybe not essential to untangle that right now.

Reply:
✅ [`056b6b5..232379d`](https://github.com/mbta/screens/pull/2061/files/056b6b57066a5fb1e14a28f15d6d1c48aa2a7f4c..232379d324176aece6ddefd06210fa3ce39da421)